Issue 446/todo app changes

ALRubinger opened the pull request:
Replacing #445 as we move this forward.

Some incremental fixes and docs updates. Dependencies on this PR ready for review:

* https://github.com/TBD54566975/incubating-web5-labs/issues/49
* https://github.com/TBD54566975/web5-tutorials/issues/7
* Whatever docs changes we need to make here which incorporate the fixes from the above release:

ALRubinger commented:
@angiejones This is now ready for your content review. Following the instructions in the guide I got this working, and it's updated to match the latest `incubating-web5-labs` tag `0.1.0` functional example that runs on Web5 `0.6.3`.

Please check any assumptions I've made in how easy it is to follow, and you may want to give specific focus to the prose pieces explaining the code to ensure they're clear. I focused primarily on getting everything working. :)

With your +1 on review, please send to @dayhaysoos for merge. 

@dayhaysoos, you may want to put this in `main` because it reflects an update to bring the TODO tutorial and its supporting finished product (`incubating-web5-labs`) and starter template (`web5-templates`) in line with Web5 `0.6.3`. It does not include the final Phase 4 update, which will hopefully have minimal changes and come later this week after Web5 `0.7.0` drops. Or, you may want to put this PR into `phase-4` branch. Your call.

Thank you all for your help in getting this and all its dependencies sorted! Props @bobbilee19 @kirahsapong @dayhaysoos @frankhinek @michaelneale.

angiejones commented on docs/web5/build/apps/todo-app-tutorial.mdx:
````diff
@@ -6,32 +6,37 @@ The ToDo app is a basic single-user, local application that only leverages `web5
 ```bash
 git clone https://github.com/TBD54566975/web5-tutorials.git
+cd web5-tutorials
+git checkout tags/0.1.0 -b todo-app-tutorial
````
@chris-tbd note versioning tag

ALRubinger replied:
Yeah there are at least 3 versioned references in here :) We’ll prob need to inventory them.

angiejones commented on docs/web5/build/apps/todo-app-tutorial.mdx:
````diff
@@ -61,98 +65,94 @@ Because we’ve built this sample using `Vue.js`, you’ll find both the HTML st
+```js
+  // ... Below code from "onBeforeMount" from above
+  
+  // Add entry to Todo array
````
`todo`

ALRubinger replied:
Addressed all caps strategy for ToDo in prose and `todo` in code according to @angiejones [guidance](https://discord.com/channels/937858703112155166/1104126737199136939/1105632989918019584): 

[e7a56d2]

angiejones commented on docs/web5/build/apps/todo-app-tutorial.mdx:
````diff
@@ -61,98 +65,94 @@ Because we’ve built this sample using `Vue.js`, you’ll find both the HTML st
 ```
 
-Now that your todos are stored in your app, you can display them by using your `todos` object. In this example, you can do it by adding the following HTML code to the Vue app below the JS, just as you would in any non-web5 app:
+Now that your todos are stored in your app, you can display them by using your `todos` object. In this example, you can do it by adding the following HTML code to the Vue app below the JS, just as you would in any non-web5 app. In our template there is a comment marked `<!-- Todos -->` where this code block fits nicely:
````
In ~our~ the template

ALRubinger replied:
Corrected all "our" instances I could find to "your" or "the": 19521d3

angiejones commented on docs/web5/build/apps/todo-app-tutorial.mdx:
```diff
@@ -161,166 +161,130 @@ Note that this UI code fulfills more than just loading the todos - it also provi
 ## Adding Todos
 
-To allow for add functionality, begin by creating the UI needed to add todos:
+To allow for add functionality, begin by creating the UI needed to add todos. In our template there is a section marked `<!-- Add Todo Form -->` where this block may go:
```
In ~our~ the template

ALRubinger replied:
Corrected all "our" instances I could find to "your" or "the": 19521d3

ALRubinger commented on docs/web5/build/apps/todo-app-tutorial.mdx:
```diff
@@ -42,8 +47,7 @@ npm install
-You should now have the app running on [`https://localhost:5173`](https://localhost:5173/) , but should see errors since your `App.vue` is not yet completely implemented.
-
+You should now have the app running on [`https://localhost:5173`](https://localhost:5173/); this will be a starter application with a mostly blank page. In this tutorial, we'll build the rest. 
 
```
Correct; no more console errors since the last upgrade. Really appreciate your walking through all these changes to make sure I didn't accidentally remove something we needed.

ALRubinger commented on docs/web5/build/apps/todo-app-tutorial.mdx:
````diff
@@ -61,98 +65,94 @@ Because we’ve built this sample using `Vue.js`, you’ll find both the HTML st
 
-```jsx
-const dwnDID = ref('');
-const web5 = new Web5();
````
Agree. That's literally how we instantiate Web5 - the object through which all this works.

ALRubinger replied:
Done in 5bce8ea.

ALRubinger commented on docs/web5/build/apps/todo-app-tutorial.mdx:
```diff
@@ -61,98 +65,94 @@ Because we’ve built this sample using `Vue.js`, you’ll find both the HTML st
-
   // Load DWN DID from local storage or create a new one
-  if(localStorage.getItem('dwn-info')){
+  if(localStorage.getItem('registerInfo') && localStorage.getItem('myDid')){
```
@kirahsapong I think can provide context behind this change; she and @bobbilee19 worked to rev the `incubating-web5-labs` one to make this work based on Frank's guidance.

angiejones replied:
sorry my comment is about the name of the dwn object in localStorage. it was `dwn-info` and is now `registerInfo`

ALRubinger replied:
> Ultimately I'd probably recommend _not_ storing `registerInfo` in localStorage or in any variable, and just passing it directly into the `web.did.manager.set()` method as a second argument. So something like this

> That said, it definitely works as is, too!

@kirahsapong Thanks! How about I open an issue for a new version of `incubating-web5-labs` where we clean this up as you suggest, and upgrade here separately later? Looking to limit impact on changes now that things meet the minimum bar - working product :P

ALRubinger replied:
Leaving this for now. Opened https://github.com/TBD54566975/developer.tbd.website/issues/455 to address.

ALRubinger commented on docs/web5/build/apps/todo-app-tutorial.mdx:
````diff
@@ -61,98 +65,94 @@ Because we’ve built this sample using `Vue.js`, you’ll find both the HTML st
-onMounted(async () => {
+```js
+onBeforeMount(async () => {
   let registerInfo;
````
Explained in tutorial as part of 5bce8ea.

ALRubinger commented on docs/web5/build/apps/todo-app-tutorial.mdx:
````diff
@@ -61,98 +65,94 @@ Because we’ve built this sample using `Vue.js`, you’ll find both the HTML st
-    todo.data = JSON.parse(todoStringified);
-    storedTodos.push(todo);
+```js
+  // ... Below code from "onBeforeMount" from above
````
Done in 1a249e0.

ALRubinger commented:
Addressed all of @angiejones's excellent feedback in separate commits on this PR, each linked to the code where she placed a comment. Updated `web5-tutorials` to match requirements, released it as `0.1.1`, and upgraded here. Ran another QA round as a user and caught some fixes to make. 

I believe this is good enough to go to `main`; subsequent issues found can be handled then.

This completes the upgrade of the ToDo tutorial, the finished example in `incubating-web5-labs` as `0.1.0`, the starter template in `web5-tutorials` as `0.1.1`, all using Web5 `0.6.3`.

This clears @bobbilee19 to have a smoother path when upgrading the full docs set to Web5 `0.7.0` or whatever version is next.

ALRubinger commented on docs/web5/build/apps/todo-app-tutorial.mdx:
````diff
@@ -61,98 +65,94 @@ Because we’ve built this sample using `Vue.js`, you’ll find both the HTML st
+```js
+  // ... Below code from "onBeforeMount" from above
+
+  // Populate Todos from DWN
````
Addressed all caps strategy for ToDo in prose and todo in code according to @angiejones [guidance](https://discord.com/channels/937858703112155166/1104126737199136939/1105632989918019584):

[https://github.com/TBD54566975/developer.tbd.website/commit/e7a56d2859b81d0a0766d49f79aaf0f33ebdb1d8]
